src/System.Private.DataContractSerialization/src/System/Xml/XmlBaseReader.cs

```diff
@@ -832,6 +832,12 @@ public override XmlNodeType MoveToContent()
                 {
                     if (_node.NodeType != XmlNodeType.Text && _node.NodeType != XmlNodeType.CDATA)
                         break;
+
+                    if (_trailByteCount > 0)
+                    {
+                        break;
+                    }
+
```

Comment:
This code already exists in the desktop code - any idea how this chunk of code disappeared from the CoreFX implementation? I want to make sure we haven't accidentally backed out a fix for something else, and Git blame isn't useful here as it was part of the initial commit. 

http://referencesource.microsoft.com/#System.Runtime.Serialization/System/Xml/XmlBaseReader.cs,858